test/oauth-authorization-test.js

```diff
@@ -335,24 +351,38 @@ suite.addBatch({
                         requestToken(cl, this);
                     },
                     function(err, rt) {
+                        var self = this;
+
                         if (err) throw err;
-                        br = new Browser({runScripts: false});
-                        br.visit("http://localhost:4815/oauth/authorize?oauth_token=" + rt.token, this);
+                        br = new Browser({runScripts: false}); //faster without scripts...
+                        Step(
+                            function(){
+                        	    br.visit("http://localhost:4815/oauth/authorize?oauth_token=" + rt.token, this);
+                            },
+                            function(){
+                                self(!br.success,br);
+                            }
+                        );
                     },
                     function(err, br) {
+                        var self = this;
+
                         if (err) throw err;
-                        if (!br.success) throw new Error("Browser error");
-                        br.fill("username", "dormouse", this);
-                    },
-                    function(err) {
-                        if (err) throw err;
-                        br.fill("password", "BADPASSWORD", this);
-                    },
-                    function(err) {
-                        if (err) throw err;
-                        br.pressButton("#authenticate", this);
+                        br.assert.success();
+
+                        br
+                            .fill("username", "dormouse", this)
```

Comment:
I don't know if jscs can handle this. Indentation needs to be 4 spaces.

With 
```
br.fill(a)
  .funky(a,function(){
      var identHowMuch = lastBrIndent+4; // ?
  });
```
the indentation when nesting a function expression is kind off weird.
So I prefer newline with one indent.
```
br
    .fill(a)
    .funky(a,function(){
        var identHowMuch = lastBrIndent+4; // ?
    });
```
The bad would be this indent has no closing bracket or something.
```
if(true){
    object
        .dothis()
        .dothat(function(){
            nop();
        });
}
```
There is a drop of 8 spaces  after `});` that's a kind of weird, but thats also (even more) the case with using the `newline indent objectname.length`, you'll get a variable drop  and not a 4 or 8 space drop.
A code style could be to close a method chain with a  comment:
```
if(true){
    object
        .dothis()
        .dothat(function(){
            nop();
        });
    // end chain
}
```


[this project](https://github.com/DarkPark/jscs) also uses a new line (search for method chain) for method chaining. [Stackoverflow](http://stackoverflow.com/questions/29784473/linting-promises-in-javascript) has some info on this (hmm not all related but got the identation). There an [issue](https://github.com/jscs-dev/node-jscs/issues/1736) at jscs-dev.

For method chains (and promises) I like the chaining to begin on a new line. Its kind of  c++ doing `using namespace obj` only there you don't need a period to use the function.

But feel free to disagree. 
I've seen http://javascript.info/tutorial/with-operator#why-with-is-deprecated . Using this analogy the example code would be:
```
br.fill(a)
br.funky(a,function(){
    var identHowMuch = lastBrIndent+4; // ?
 });
```